### PR TITLE
Update consensus to v0.7.0

### PR DESCRIPTION
**Pull request by mkalinin**
### What was done?
Updates consensus to v0.7.1. Among other things are:
- Deposit logic update related to https://github.com/ethereum/eth2.0-specs/pull/1152

_Note:_ moved `ChainStart` class to `consensus` module to get rid of circular dependency. Not a perfect solution. We may want to re-design our modules later on.

### CODE REVIEW

**Comment by mkalinin** [comment]
**Note:** part of state and ssz tests are ignored and requires and update.

**Review by Nashatyrev — APPROVED** [review]

File: core/src/main/java/org/ethereum/beacon/core/operations/attestation/Crosslink.java [hunk 1]
```diff
@@ -12,68 +12,87 @@
...
-  public static final Crosslink EMPTY = new Crosslink(EpochNumber.ZERO, Hash32.ZERO, Hash32.ZERO);
+  public static final Crosslink EMPTY =
+      new Crosslink(ShardNumber.ZERO, EpochNumber.ZERO, EpochNumber.ZERO, Hash32.ZERO, Hash32.ZERO);
 
-  /** Epoch number. */
-  @SSZ private final EpochNumber epoch;
+  /** Shard number. */
+  @SSZ private final ShardNumber shard;
+  /** Crosslinking data from epochs [start....end-1]. */
+   @SSZ private final EpochNumber startEpoch;
```
Comment (zilm13) [inline core/src/main/java/org/ethereum/beacon/core/operations/attestation/Crosslink.java]: Space!
```suggestion
@SSZ private final EpochNumber startEpoch;
```

File: validator/src/main/java/org/ethereum/beacon/validator/proposer/BeaconChainProposerImpl.java [hunk 2]
```diff
@@ -155,6 +155,7 @@ private Eth1Data getEth1Data(BeaconState state) {
         .filter(
             eth1Data ->
                 depositContract.hasDepositRoot(eth1Data.getBlockHash(), eth1Data.getDepositRoot()))
+        // TODO throw exception if contract data can't be red
```
Comment (zilm13) [inline validator/src/main/java/org/ethereum/beacon/validator/proposer/BeaconChainProposerImpl.java]:
```suggestion
        // TODO throw exception if contract data can't be read
```

**Review by zilm13 — CHANGES_REQUESTED** [review]
I'm not sure you are able to deal with it

**Pull request merged**